lib/katello/tasks/test.rake

```diff
@@ -42,6 +42,8 @@ namespace :test do
       test_task = Rake::TestTask.new('katello_test_task') do |t|
         t.libs << ["test", "#{Katello::Engine.root}/test"]
         t.test_files = [
+          "#{Katello::Engine.root}/test/models/**/*_test.rb",
+          "#{Katello::Engine.root}/test/controllers/**/*_test.rb",
```

Comment:
This is something @jlsherrill and I looked into at length and is still an outstanding issue. At some point throughout testing the permissions would stop loading and we would around 14 errors such as

```Katello::RepositoryAuthorizationNonAuthUserTest#test_readable_with_product:
RuntimeError: Permissions not found: view_products, some permissions were not found:  ["view_architectures", "create_architectures", "edit_architectures", "destroy_architectures", "view_authenticators", "create_authenticators", "edit_authenticators", "destroy_authenticators", "view_bookmarks", "create_bookmarks", "edit_bookmarks", "destroy_bookmarks", "view_compute_resources", "create_compute_resources", "edit_compute_resources", "destroy_compute_resources", "view_templates", "create_templates", "edit_templates", "destroy_templates", "deploy_templates", "access_dashboard", "view_domains", "create_domains", "edit_domains", "destroy_domains", "view_realms", "create_realms", "edit_realms", "destroy_realms", "view_environments", "create_environments", "edit_environments", "destroy_environments", "import_environments", "view_external_variables", "create_external_variables", "edit_external_variables", "destroy_external_variables", "view_globals", "create_globals", "edit_globals", "destroy_globals", "create_params", "edit_params", "destroy_params", "view_hostgroups", "create_hostgroups", "edit_hostgroups", "destroy_hostgroups", "view_hosts", "create_hosts", "edit_hosts", "destroy_hosts", "build_hosts", "power_hosts", "console_hosts", "ipmi_boot", "puppetrun_hosts", "view_images", "create_images", "edit_images", "destroy_images", "view_locations", "create_locations", "edit_locations", "destroy_locations", "assign_locations", "view_media", "create_media", "edit_media", "destroy_media", "view_models", "create_models", "edit_models", "destroy_models", "view_operatingsystems", "create_operatingsystems", "edit_operatingsystems", "destroy_operatingsystems", "view_organizations", "create_organizations", "edit_organizations", "destroy_organizations", "assign_organizations", "view_ptables", "create_ptables", "edit_ptables", "destroy_ptables", "view_puppetclasses", "create_puppetclasses", "edit_puppetclasses", "destroy_puppetclasses", "import_puppetclasses", "view_reports", "destroy_reports", "upload_reports", "access_settings", "view_smart_proxies", "create_smart_proxies", "edit_smart_proxies", "destroy_smart_proxies", "view_smart_proxies_autosign", "create_smart_proxies_autosign", "destroy_smart_proxies_autosign", "view_smart_proxies_puppetca", "edit_smart_proxies_puppetca", "destroy_smart_proxies_puppetca", "view_statistics", "view_subnets", "create_subnets", "edit_subnets", "destroy_subnets", "import_subnets", "view_tasks", "view_trends", "create_trends", "edit_trends", "destroy_trends", "update_trends", "view_usergroups", "create_usergroups", "edit_usergroups", "destroy_usergroups", "view_users", "create_users", "edit_users", "destroy_users", "edit_classes", "view_compute_resources_vms", "create_compute_resources_vms", "edit_compute_resources_vms", "destroy_compute_resources_vms", "power_compute_resources_vms", "console_compute_resources_vms", "view_facts", "upload_facts", "view_audit_logs", "view_plugins", "view_provisioning_templates", "view_roles"]
    /home/vagrant/katello/test/support/auth_support.rb:13:in `rescue in block in create_role_with_permissions'
    /home/vagrant/katello/test/support/auth_support.rb:8:in `block in create_role_with_permissions'
    /home/vagrant/katello/test/support/auth_support.rb:7:in `each'
    /home/vagrant/katello/test/support/auth_support.rb:7:in `create_role_with_permissions'
    /home/vagrant/katello/test/support/auth_support.rb:38:in `setup_user_with_permissions'
    /home/vagrant/katello/test/support/auth_support.rb:45:in `setup_current_user_with_permissions'
    /home/vagrant/katello/test/models/authorization/repository_authorization_test.rb:69:in `test_readable_with_product'
```

the point that the permissions stop loading seems to be completely random and not related to any test, and the only fix we have come up with is to make sure to run the tests that will fail without katello permissions first